src/main/java/frc/robot/subsystems/MissileShellVisionSubsystem.java

```diff
@@ -0,0 +1,31 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import frc.robot.vision.PhotonVision;
+
+public class MissileShellVisionSubsystem extends SubsystemBase {
+    private final PhotonVision vision;
+    private Pose2d pose = new Pose2d();
+
+    private final ShuffleboardTab shuffleboardTab = Shuffleboard.getTab("Vision Testing");
+    private final GenericEntry xEntry = shuffleboardTab.add("Pose x", 0.0).getEntry();
+    private final GenericEntry yEntry = shuffleboardTab.add("Pose y", 0.0).getEntry();
+
+    public MissileShellVisionSubsystem(PhotonVision vision) {
+        this.vision = vision;
+    }
+
+    @Override
+    public void periodic() {
+        var res = vision.getRobotPose(pose);
+        this.pose = res.getFirst();
+
+        xEntry.setDouble(pose.getX());
+        yEntry.setDouble(pose.getY());
```

Comment:
oo shuffleboard is pretty easy

let's add the robot angle too?

Reply:
Perhaps? I deleted the test class entirely because I wasn't sure if the info it was getting was valuable at all. The vision pose estimator needs a previous reference pose to work, but it seems like passing its old estimate back to it creates some sort of feedback loop where it just gives the same position over and over? Either that or there was something off with the camera position or how it was mounted; I think it might be easier to mount it on an actual swerve base with odometry to test.

Reply:
this seems really critical to debug even before running on an actual swerve base. 
are vision estimates currently passed to the poseestimator even if they're null? what's the behavior of doing so